lib/cluster/command.js

```diff
@@ -0,0 +1,262 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2020 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const { Reply, Request } = require('zeromq');
+const protobuf = require('protobufjs');
+const Bluebird = require('bluebird');
+
+/* eslint-disable sort-keys */
+const stateEnum = Object.freeze({
+  INITIALIZING: 1,
+  RUNNING: 2,
+  CLOSED: 3,
+});
+
+const topicEnum = Object.freeze({
+  FULLSTATE: 'fullstate',
+  HANDSHAKE: 'handshake',
+  DISCARDED: 'discarded', // 0 byte message expected
+});
+/* eslint-enable sort-keys */
+
+// Handles bidirectional requests between this node and other nodes
+class ClusterCommand {
+  /**
+   * @constructor
+   * @param {ClusterNode} localNode
+   */
+  constructor (localNode) {
+    this.node = localNode;
+    this.server = null;
+    this.protoroot = null;
+
+    this.state = stateEnum.INITIALIZING;
+  }
+
+  async init () {
+    this.protoroot = await protobuf.load(`${__dirname}/protobuf/command.proto`);
+    this.serve();
```

Comment:
This can return a rejected promise

Reply:
Ah, yes, when binding to the network port for instance. I'll have to separate the socket creation and its listener.